### PR TITLE
[FIX] point_of_sale: Avoid errors while accessing a point of sale ses…

### PR DESCRIPTION
**Pull request by hubvd**
…sion while it's pricelist was archived.

Show a UserError while archiving a point of sale pricelist if it is being used a as default pricelist for a point of sale.

Ticket: 2463649

### CODE REVIEW

File: addons/point_of_sale/models/product_pricelist.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from odoo import models, _
+from odoo.exceptions import UserError
+
+
+class PosPayment(models.Model):
```
Comment (pimodoo) [inline addons/point_of_sale/models/product_pricelist.py:5]: the name of the class is wrong

File: addons/point_of_sale/models/product_pricelist.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    _inherit = "product.pricelist"
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'active' in vals and not vals['active'] and self.env['pos.config'].sudo().with_context(active_test=False).search_count([('pricelist_id', 'in', self.ids)]):
```
Comment (pimodoo) [inline addons/point_of_sale/models/product_pricelist.py:10]: do your stuff before the call of super, as it'll avoid to write and rollback it afterward

**Comment by pimodoo** [comment]
@hubvd You are breaking a lot of tests on the runbot, so not sure you want to fix them for the added value of this fix.

**Comment by hubvd** [comment]
@pimodoo I managed to fix the tests, but maybe this is a bit too hacky. What do you think ?

**Comment by hubvd** [comment]
ping @fmdl

**Pull request closed without merging**